CHANGES.md

```diff
@@ -1,6 +1,417 @@
+* renamed search arguments `needle`->`pattern` and `region`->`indices`  
+  (see https://chapel-lang.org/docs/1.26/builtins/Bytes.html,  
+   https://chapel-lang.org/docs/1.26/builtins/String.html,  
+   and https://chapel-lang.org/docs/1.26/modules/standard/Regex.html)
+* renamed `regexMatch.offset` to `.byteOffset` and `.size` to `.numBytes`  
+  (see https://chapel-lang.org/docs/1.26/modules/standard/Regex.html#Regex.regexMatch)
+* renamed `captures`/`maxmatches` args in 'Regex' to `numCaptures`/`maxMatches`
+* renamed the `list.indexOf()` method to `list.find()`  
+  (see https://chapel-lang.org/docs/1.26/modules/standard/List.html#List.list.indexOf)
+* renamed several `set` method arguments from `x` to `element`  
+  (see https://chapel-lang.org/docs/1.26/modules/standard/Set.html)
+* changed the formal names in the `Reflection.getField()` family of functions  
+  (see https://chapel-lang.org/docs/1.26/modules/standard/Reflection.html)
+* renamed `Subprocess.send_signal()` to `Subprocess.sendPosixSignal()`  
+  (see https://chapel-lang.org/docs/1.26/modules/standard/Subprocess.html#Subprocess.subprocess.sendPosixSignal)
+* changed 'Subprocess' pipe styles from a set of constants to an `enum`  
+  (see https://chapel-lang.org/docs/1.26/modules/standard/Subprocess.html#Subprocess.pipeStyle)
+* improved the names of many routines and arguments in the 'BigInteger' module
+  (see https://chapel-lang.org/docs/1.26/modules/standard/BigInteger.html)
+  - renamed `bigint.mpzStruct()` to `bigint.getImpl()`
+  - renamed `bigint.get_d_2exp()` to `bigint.getD2Exp()`
+  - renamed `bigint.divisible_p()` to `bigint.isDivisible()`
+  - renamed `bigint.divisible_2exp_p()` to `bigint.isDivisibleBy2Pow()`
+  - renamed `bigint.congruent_p()` to `bigint.isCongruent()`
+  - renamed `bigint.congruent_2exp_p()` to `bigint.isCongruentBy2Pow()`
+  - renamed `bigint.perfect_power_p()` to `bigint.isPerfectPower()`
+  - renamed `bigint.perfect_square_p()` to `bigint.isPerfectSquare()`
+  - renamed `bigint.probab_prime_p()` to `bigint.probablyPrime()`
+  - renamed `bigint.gcdext()` to new `bigint.gcd()` overload
+  - renamed `bigint.remove()` to `bigint.removeFactor()`
+  - renamed `bigint.even_p()` to `bigint.isEven()`
+  - renamed `bigint.odd_p()` to `bigint.isOdd()`
+  - renamed arguments of `bigint.divexact()` to `numer` and `denom`
+  - renamed argument of `bigint.isDivisible()` from `d` to `div`
+  - renamed argument of `bigint.isDivisibleBy2Pow()` from `b` to `exp`
+  - renamed arguments of `bigint.isCongruent()` to `con` and `mod`
+  - renamed arguments of `bigint.isCongruentBy2Pow()` to `con` and `modExp`
+  - renamed argument of `bigint.scan0()` and `bigint.scan1()` to `startBitIdx`
+
+Deprecated / Removed Library Features
+-------------------------------------
+* deprecated the automatic 'ChapelEnv' module for 'ChplConfig'  
+  (see https://chapel-lang.org/docs/1.26/modules/standard/ChplConfig.html)
```

Comment:
I wasn't sure if this sort of thing made more sense here or in "Name Changes in Libraries"

Reply:
There are some other ones that kinda feel like name changes. I don't think worth changing, but I was wondering if we should have a single place users can look to see what all names changes took place (like library evolution or something, but the CHANGES file feels like the wrong place for that, and it seems like more work than it's worth especially since we issue deprecation warnings.)

Reply:
You're definitely right that there's a grey line here...  I think since this one was a name that most people probably don't know, use, or refer to, I thought it might be better to put it here, but it could go in either category.

> especially since we issue deprecation warnings

That's my thinking.  Seems like nobody would read the docs wherever they lived until they were surprised, and when we do good warnings, that seems like the better investment in time.